src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -395,12 +476,16 @@ private static String getClassNameFromQualifiedName(String qualifiedName) {
          *         class declared as final
          *  @param declaredAsAbstract indicates if the
          *         class declared as abstract
+         *  @param depth class nesting level
+         *  @param classAst classAst node
          */
         /* package */ ClassDesc(String qualifiedName, boolean declaredAsFinal,
-                boolean declaredAsAbstract) {
+                boolean declaredAsAbstract, int depth, DetailAST classAst) {
```

Comment:
Now that we pass in the class AST, can we just eliminate the two boolean parameters (`declaredAsFinal`, `declaredAsAbstract`) and check for these in the constructor?

Reply:
done

Reply:
Resolved.